packages/transformers/babel/src/config.js

```diff
@@ -1,141 +1,191 @@
-  let pkg = await asset.getPackage();
-  let isSource =
-    !!(
-      pkg &&
-      pkg.source &&
-      (await asset.fs.realpath(asset.filePath)) !== asset.filePath
-    ) || !asset.filePath.includes(NODE_MODULES);
-
-  // Try to resolve a .babelrc file. If one is found, consider the module source code.
-  let babelrc = await getBabelRc(asset, pkg, isSource);
-  isSource = isSource || !!babelrc;
-
-  let result = {};
-  mergeConfigs(result, babelrc);
-
-  // Typescript must use the plugin directly (not the typescript preset) and must
-  // come before preset env, otherwise proposals and nonstandard syntax is not
-  // transformed in time.
-  if (path.extname(asset.filePath).match(TYPESCRIPT_EXTNAME_RE)) {
-    let hasTypescript =
-      babelrc &&
-      (hasPlugin(babelrc.config.presets, [
-        '@babel/typescript',
-        '@babel/preset-typescript'
-      ]) ||
-        hasPlugin(babelrc.config.plugins, [
-          '@babel/transform-typescript',
-          '@babel/plugin-transform-typescript'
-        ]));
-
-    if (!hasTypescript) {
-      mergeConfigs(result, {
-        internal: true,
-        babelVersion: 7,
-        config: {
-          plugins: [
-            [
-              require('@babel/plugin-transform-typescript'),
-              {isTSX: path.extname(asset.filePath) === '.tsx'}
-            ]
-          ]
-        }
+const BABEL_TRANSFORMER_DIR = path.dirname(__dirname);
+
+export async function load(config: IConfig) {
+  let partialConfig = loadPartialConfig({filename: config.searchPath});
+  if (partialConfig && partialConfig.hasFilesystemConfig()) {
+    let {babelrc, config: configjs} = partialConfig;
+    let {canBeRehydrated, dependsOnRelative, dependsOnLocal} = getStats(
+      partialConfig.options
+    );
+
+    // babel.config.js files get required by @babel/core so there's no use in including it for watch mode invalidation
+    if (babelrc != null && configjs == null) {
+      config.setResolvedPath(babelrc);
+    } else if (configjs) {
+      // TODO: warn and invalidate on startup
+    }
+
+    if (babelrc && (await isExtended(/*babelrc*/))) {
+      // TODO: warn and invalidate on startup
+      // TODO: maybe add feature to babel that gives details on extended files?
+    }
+
+    if (dependsOnRelative || dependsOnLocal) {
+      // TODO: warn that you may not see changes to relative or local plugins/presets until their
+      // containing package changes versions. You can run with --no-cache to see changes until ready
+      // to bump version.
+    }
+
+    if (canBeRehydrated) {
+      prepForReyhdration(partialConfig.options);
+      config.shouldRehydrate();
+      config.setResult({
+        internal: false,
+        config: partialConfig.options
       });
+
+      await definePluginDependencies(config);
+      config.setResultHash(md5FromObject(partialConfig.options));
+    } else {
+      config.shouldReload();
+      config.setResultHash(JSON.stringify(Date.now()));
```

Comment:
is a hash necessary in this case?

Reply:
Well the hash needs to be different each time because we have no way of knowing if the config is the same as last time or not. Are you suggesting something like core doing this automatically is no hash is set?

Reply:
Yeah I was kinda thinking of not having plugins set hashes at all, and we'd just do it in core...